### ISSUE TITLE
opaque error with kml downloaded from Google My Maps

### ISSUE THREAD

**Issue by yeedle** [r-spatial/mapview#98]
I'm not sure where to raise this issue, so I'm raising this here.

When trying to view [this](http://www.nhc.noaa.gov/gis/kml/surge/WatchWarningSS_latest.kml) data with mapview, I kept on getting a weird error:

```r
Error in if (length(nms) != n || any(nms == "")) stop("'options' must be a fully named list, or have no names (NULL)") : 
  missing value where TRUE/FALSE needed
```
After some head scratching I figured out that KML files that are made with [Google My Maps](https://www.google.com/maps/about/mymaps/) and perhaps other map creation software as well, come by default with a z dimension set to 0, and this causes the error when trying to view it with mapview and leaflet. After removing the z dimension with `st_zm` I was able to view it without any problems.

Perhaps a check can be made for a z dimension, and in such a case a more helpful error message can be given. At the very least, mention of this in the documentation somewhere might save someone some head-scratching time in the future. I don't mind doing a PR for this if it's not a problem.

**Comment by tim-salabim** [comment]
Thanks for this! Finally a use-case with "real-world" for `XYZ` coordinate data. 

I think we can easily handle this in a couple of ways. Right from the top of my head I think I'd prefer to define a method for `XYZ` which automatically matches z-values to `zcol` (so finally the name actually makes sense). Probably the same should be done for `XYM` but I am not sure about a best practice for `XYZM` (all of which are supported by `sf`). 

Let me think about this a little more.

**Comment by tim-salabim** [comment]
https://github.com/r-spatial/mapview/commit/c41df1ec34d1cdc4bf9623f55ed8979dd5f1dd89#diff-b8255b930797229b81d94ca8229302f6R91 should fix this. 

I opted for the simplest possible solution to simply drop `ZM` via `st_zm` as automatic mapping of `ZM` values will only be possible for point data. Thus dropping these coordinates for all geometry types seems to be most coherent. I added a note in the details of `?mapview`
